oemoflex/model/postprocessing.py

```diff
@@ -789,20 +789,41 @@ def run_postprocessing(es):
         summed_marginal_costs,
     ]
 
-    all_scalars = pd.concat(all_scalars, axis=0)
+    # all_scalars = pd.concat(all_scalars, axis=0)
+    # This does not work with the update from pandas==2.0.3 to pandas==2.2.1 because
+    # invested_capacity and invested_storage_capacity both have a TimeStamp as column name which
+    # results in a mix-up of the levels
+    # Fixing Approach:
+    # list(map(lambda series: series.rename('0', inplace=True), all_scalars))
+    # timestamp_variable = pd.to_datetime("2017-01-01 00:00:00")
+    # all_scalars = pd.concat(all_scalars, axis=0,
+    # keys=['source', 'target', 'var_name', '0', 0, 'var_value', timestamp_variable])
+    # did not work
+    # Todo: To be further investigated
+
+    # Index work-around - issues with concat and Multiindex
+    list(map(lambda series: series.rename("0", inplace=True), all_scalars))
```

Comment:
This is not recommended, as you are abusing list generation to perform actions on a series (see i.e. https://stackoverflow.com/a/5753614/5804947).
Instead you could write:
`all_scalars_renamed = [series.rename("0") for series in all_scalars]`

Reply:
or merge this directly with next call:
`all_scalars_reindexed = [series.rename("0").reset_index() for series in all_scalars]`

Reply:
Unfortunately I get the following error with https://github.com/rl-institut/oemoflex/commit/a31f733d133019decde3ca0b73897821572e17c6:

```
INFO - 
Traceback (most recent call last):
  File "~/Schreibtisch/oemof-B3/scripts/plot_scalar_results.py", line 524, in <module>
    scalars = load_scalar_results(scalars_path)
  File "~/Schreibtisch/oemof-B3/scripts/plot_scalar_results.py", line 502, in load_scalar_results
    df = dp.format_header(
  File "~/Schreibtisch/oemof-B3/oemof_b3/tools/data_processing.py", line 103, in format_header
    raise ValueError(f"There are extra columns {extra_colums}")
ValueError: There are extra columns ['0']

```

Reply:
It is fixed with commit https://github.com/rl-institut/oemoflex/pull/100/commits/0fc2956b633a595540d1cbb7ffba4f4ec1d41e27, but the DataFrame all_scalars from the version before the pandas update is now different from the current dataframe. However, the results of the postprocessing seem to be the same. I have compared the scalars.csv files with each other.